packages/react-components/react-positioning/src/usePositioning.ts

```diff
@@ -300,90 +45,63 @@ export function usePositioning(
-
-  const popperInstanceRef = React.useRef<PopperInstance | null>(null);
-
-  const handlePopperUpdate = useEventCallback(() => {
-    popperInstanceRef.current?.destroy();
-    popperInstanceRef.current = null;
+  const resolvePositioningOptions = usePositioningOptions(options);
 
+  const forceUpdate = useEventCallback(() => {
     const target = overrideTargetRef.current ?? targetRef.current;
-
-    let popperInstance: PopperInstance | null = null;
-
-    if (canUseDOM() && enabled) {
-      if (target && containerRef.current) {
-        popperInstance = PopperJs.createPopper(
-          target,
-          containerRef.current,
-          resolvePopperOptions(target, containerRef.current, arrowRef.current),
-        );
-      }
+    if (!canUseDOM || !enabled || !target || !containerRef.current) {
+      return;
     }
 
-    if (popperInstance) {
-      /**
-       * The patch updates `.forceUpdate()` Popper function which restores the original position before the first
-       * forceUpdate() call. See also "positionStyleFix" modifier in usePopperOptions().
-       */
-      const originalForceUpdate = popperInstance.forceUpdate;
-
-      popperInstance.isFirstRun = true;
-      popperInstance.forceUpdate = () => {
-        if (popperInstance?.isFirstRun) {
-          popperInstance.state.elements.popper.style.position = popperOriginalPositionRef.current;
-          popperInstance.isFirstRun = false;
+    const { placement, middleware, strategy } = resolvePositioningOptions(
+      target,
+      containerRef.current,
+      arrowRef.current,
+    );
+
+    // Container is always initialized with `position: fixed` to avoid scroll jumps
+    // Before computing the positioned coordinates, revert the container to the deisred positioning strategy
+    Object.assign(containerRef.current.style, { position: strategy });
+    computePosition(target, containerRef.current, { placement, middleware, strategy })
+      .then(({ x, y, middlewareData, placement: computedPlacement }) => {
+        writeArrowUpdates({ arrow: arrowRef.current, middlewareData });
+        writeContainerUpdates({
+          container: containerRef.current,
+          middlewareData,
+          placement: computedPlacement,
+          coordinates: { x, y },
+          lowPPI: (targetDocument?.defaultView?.devicePixelRatio || 1) <= 1,
+          strategy,
+        });
+      })
+      .catch(err => {
+        // This is super duper stuuuuuuupid
+        // FIXME for node > 14
+        // node 15 introduces promise rejection which means that any components
+        // tests need to be `it('', async () => {})` otherwise there can be race conditions with
+        // JSOM being torn down before this promise is resolved.
+        // Unless all tests that ever use `usePositioning` are turned into async tests, any logging during testing
+        // will actually be counter productive
+        if (process.env.NODE_ENV === 'development') {
```

Comment:
shouldn't we use check as we do in other places ?
```suggestion
        if (process.env.NODE_ENV !== 'production') {
```

Reply:
see https://github.com/microsoft/fluentui/pull/24254#discussion_r940160910